vowpalwabbit/options.h

```diff
@@ -88,25 +91,36 @@ struct typed_option : base_option
   typed_option& value(T value)
   {
     m_value = std::make_shared<T>(value);
+    if (m_location != nullptr) { *m_location = value; }
+    value_set_callback(value);

```

Comment:
so the callback gets called always, it is not an either or situation depending on if m_location is set?

Reply:
The callback is not related to the location, so it is always called.